src/ErrorHandler.php

```diff
@@ -283,31 +257,17 @@ public function handleException(\Throwable $exception): void
     }
 
     /**
-     * Re-registers the error handler if the mask that configures the intercepted
-     * error types changed.
+     * Fills a static property with a string to reserve some memory to be used while handling fatal errors.
      *
-     * @param int $previousThrownErrors The previous error mask
+     * @param int $reservedMemorySize The amount of memory to be reserved, is in char string length
      */
-    private function reRegister(int $previousThrownErrors): void
+    private static function setReservedMemory(int $reservedMemorySize): void
     {
-        if ($this->capturedErrors === $previousThrownErrors) {
-            return;
+        if ($reservedMemorySize <= 0) {
+            throw new \UnexpectedValueException('The $reservedMemorySize argument must be greater than 0.');
```

Comment:
Isn't `InvalidArgumentException` a more appropriate exception in this case (since it talks about the argument and not a generic value)?